tests/functions_tests/test_hierarchical_softmax.py

```diff
@@ -34,19 +39,40 @@ def setUp(self):
 
         self.W = self.func.W.copy()
 
-    @condition.retry(3)
-    def test_sum(self):
-        x = numpy.array([[1.0, 2.0, 3.0]], numpy.float32)
+    def check_sum(self, x, gpu=False):
         total = 0
         for i in range(5):
             t = numpy.array([i])
-            loss, = self.func.forward_cpu((x, t))
+            if gpu:
+                t = cuda.to_gpu(t)
+            loss, = self.func.forward((x, t))
             self.assertEqual(loss.dtype, numpy.float32)
             self.assertEqual(loss.shape, ())
-            total += numpy.exp(-loss)
+            total += numpy.exp(-cuda.to_cpu(loss))
         self.assertAlmostEqual(1.0, float(total), delta=1.0e-5)
 
-    def check_backward(self, x_data, t_data, y_grad, use_cudnn=True):
+    @condition.retry(3)
+    def test_sum_cpu(self):
+        x = numpy.array([[1.0, 2.0, 3.0]], numpy.float32)
+        self.check_sum(x)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_sum_gpu(self):
+        x = numpy.array([[1.0, 2.0, 3.0]], numpy.float32)
+        self.func.to_gpu()
+        self.check_sum(cuda.to_gpu(x), gpu=True)
+
+    @attr.gpu
+    def test_forwrd(self):
```

Comment:
We need unit test for forward for cpu, altough that is not the focus of this PR.